packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -531,38 +576,48 @@ describeNoCompat("GC tombstone tests", (getTestObjectProvider) => {
                 summarizingContainer,
                 summarizer,
             } = await summarizationWithUnreferencedDataStoreAfterTime(sweepTimeoutMs - remainingTimeUntilSweepMs);
-            const dataObject = await requestFluidObject<ITestDataObject>(summarizingContainer, unreferencedId);
-
             // Wait enough time so that the datastore is sweep ready
             await delay(remainingTimeUntilSweepMs);
 
             await sendOpToUpdateSummaryTimestampToNow(summarizingContainer);
 
             // The datastore should be tombstoned now
-            await summarize(summarizer);
+            const { summaryVersion } = await summarize(summarizer);
+            const tombstoneContainer = await loadContainer(summaryVersion);
+            await sendOpToUpdateSummaryTimestampToNow(tombstoneContainer);
 
-            // Note: if a user makes a request that looks like this, we will also think that the request is via handle
-            const request: IRequest = { url: unreferencedId, headers: { [RuntimeHeaders.viaHandle]: true } };
-            const response = await dataObject._context.IFluidHandleContext.resolveHandle(request);
+            // Datastore should be tombstoned
+            await assert.rejects(async () => requestFluidObject<ITestDataObject>(tombstoneContainer, unreferencedId),
+                (error) => {
+                    const correctErrorType = error.code === 404;
+                    const correctErrorMessage = error.message.startsWith(`Datastore removed by gc`) === true;
+                    return correctErrorType && correctErrorMessage;
+                },
+                `Should not be able to retrieve a tombstoned datastore.`,
+            );
 
-            assert(response !== undefined, `Expecting a response!`);
-            assert(response.status === 404);
-            assert(response.value === `Datastore removed by gc: ${unreferencedId}`);
+            const mainDataObject = await requestFluidObject<ITestDataObject>(tombstoneContainer, "default");
+            const handle = new RecoveryHandle(`/${unreferencedId}`);
```

Comment:
Why create a separate handle for this? What changed?

Reply:
1. `LegacyUriHandle` returned by `handleFromLegacyUri` would fail when it was added to a DDS because `attachGraph` would be called. -> this was an easy fix
2. The container storing the handle is already in a bad state, I didn't want the handle to actually return anything when `get` is called, so I tried having `get` just throw. 
3. Then a bunch of tests in one of our examples started failing.
4. I figured at that point, fixing those tests was not worth putting in this PR since the changes wouldn't be relevant enough to this PR.
5. So I created a handle that I thought would do what I wanted it to do
   - A recovery handle that would allow the container to save its data while in a bad state, and not exacerbate the state by preventing changes to the tombstoned datastore on that specific container.

The original way of retrieving the handle wasn't working because the data object was tombstoned (intended behaviour).

Reply:
I understand LegacyUriHandle vs the new one. What's unclear to me all of this change is needed. As in, the previous code of storing dataObject.handle would work right?
The only thing that changed is summarizer won't throw an error when tombstone is used. That should not cause any more failues (the failures should infact become fewer now)

Reply:
Hmmm, good point, the previous code would work. I can revert it. I'll add a separate test later to demonstrate recovery :).